### PR TITLE
Fixed #618

### LINKED ISSUE #618: Sub-sub folders not handled correctly
I have a subfolder of a subfolder, and it appears as 'SubFolder.SubSubFolder' instead of as a dropdown. Also it’s not loaded. Other email clients correctly deduce the sub-subfolder structure.

### PR DESCRIPTION
**Pull request by ErikPel**
Seems to do the trick.

Anyone wanna test and give feedback?

@ChristophWurst

fixes #618

### CODE REVIEW

File: js/views/navigation-accounts.js [hunk 1]
```diff
@@ -31,23 +31,17 @@ define(function(require) {
...
-				k += p;
-				var folders = activeFolder.folders || activeFolder.get('folders');
-				activeFolder = folders.filter(function(f) {
-					return f.id === btoa(k);
-				}).shift();
-			});
+
+			var first_part = atob(folderId).split(delimiter)[0];
+			activeFolder = activeAccount.get('folders').get(btoa(first_part));
+			activeFolder.attributes.folders.forEach(function(folder){
```
Comment (ChristophWurst) [inline js/views/navigation-accounts.js]: whitespace after closing bracket: ``function folder) {``

File: js/views/navigation-accounts.js [hunk 2]
```diff
@@ -31,23 +31,17 @@ define(function(require) {
...
+				if(folder.id == folderId)
```
Comment (ChristophWurst) [inline js/views/navigation-accounts.js]: left brace brace on the same line as the if

File: js/views/navigation-accounts.js [hunk 3]
```diff
@@ -31,23 +31,17 @@ define(function(require) {
...
+				{
+					activeFolder = folder;
+				} 
+			}
+			);
```
Comment (ChristophWurst) [inline js/views/navigation-accounts.js]: put the closing bracket to the line above, so it's ``});``

**Comment by ChristophWurst** [comment]
:+1: tested and works! All folders are accessible again. Thank you very much.

**Comment by ChristophWurst** [comment]
Please fix coding style error: https://travis-ci.org/owncloud/mail/jobs/106268948

Then we're ready to merge :D

**Comment by ChristophWurst** [comment]
pro tip: write "fixes #xxx" into the PR description, then github closes the issue automatically

**Comment by ErikPel** [comment]
All good now? @ChristophWurst Ready for rebase and merge?

Thanks for the tip. I'll remember that next time

File: js/views/navigation-accounts.js [hunk 4]
```diff
@@ -31,22 +31,14 @@ define(function(require) {
...
-				k += p;
-				var folders = activeFolder.folders || activeFolder.get('folders');
-				activeFolder = folders.filter(function(f) {
-					return f.id === btoa(k);
-				}).shift();
+
+			var first_part = atob(folderId).split(delimiter)[0];
+			activeFolder = activeAccount.get('folders').get(btoa(first_part));
+			activeFolder.attributes.folders.forEach(function(folder) {
+				if(folder.id == folderId) {
```
Comment (ChristophWurst) [inline js/views/navigation-accounts.js]: actually there should also be a space after the if. Simply amend your last commit.

**Comment by jancborchardt** [comment]
@ErikPel with all your cool contributions, we should add you to the ownCloud organization and the ownCloud Mail team! :)

@karlitschek this requires a signed contributor agreement as well, right? @ErikPel do you mind? https://owncloud.org/contribute/agreement/

**Comment by ErikPel** [comment]
@jancborchardt Signed it few weeks ago. Check out the comments here: 

https://github.com/owncloud/core/pull/21599#issuecomment-171309124

**Comment by jancborchardt** [comment]
@ErikPel ah cool, I added you to the `@owncloud/mail` team as well :)

You should also set your visibility from private to public at https://github.com/orgs/owncloud/people?utf8=%E2%9C%93&query=erikpel

**Pull request merged**